Compilation problem with gcc 13.2.0

Build fails with 

```
cc \
	-O2 -Wall -Werror \
	 \
	-Isrc -Ilib -I/usr/local/include -pthread \
	-I/home/phk/Varnish/trunk/varnish-cache/include \
	-o src/vtc_main.o -c src/vtc_main.c
src/vtc_main.c: In function ‘cleaner_setup’:
src/vtc_main.c:237:23: error: ignoring return value of ‘nice’ declared with attribute ‘warn_unused_result’ [-Werror=unused-result]
  237 |                 (void)nice(1);          /* Not important */
      |                       ^~~~~~~
cc1: all warnings being treated as errors
make[1]: *** [Makefile:80: src/vtc_main.o] Error 1
```

on GCC 13.2.0

Relevant GCC bug and workaround: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=66425#c34

--- a/src/vtc_main.c
+++ b/src/vtc_main.c
@@ -234,7 +234,7 @@ cleaner_setup(void)
 	assert(cleaner_pid >= 0);
 	if (cleaner_pid == 0) {
 		closefd(&p[1]);
-		(void)nice(1);		/* Not important */
+		(void)!nice(1);		/* Not important */
 		setbuf(stdin, NULL);
 		AZ(dup2(p[0], STDIN_FILENO));
 		while (fgets(buf, sizeof buf, stdin)) {

